pkg/app/piped/executor/kubernetes/rollback.go

```diff
@@ -135,9 +135,9 @@ func (e *rollbackExecutor) ensureRollback(ctx context.Context) model.StageStatus
 		return model.StageStatus_STAGE_FAILURE
 	}
 
-	cp, ok := e.PipedConfig.FindCloudProvider(e.Deployment.CloudProvider, model.ApplicationKind_KUBERNETES)
+	cp, ok := e.PipedConfig.FindPlatformProvider(e.Deployment.PlatformProvider, model.ApplicationKind_KUBERNETES)
```

Comment:
> This PR adds logic to make PlatformProviders specified via piped configuration work as the current CloudProviders. The CloudProviders configuration is marked as deprecated and will be removed after a few releases.


When reading the PR's title, I thought that this PR was only focused on the replacement in the Piped configuration.
But this PR also switches to using PlatformProvider in the model.
Is that ok? For example, the deployment still has only the CloudProvider field.

Reply:
The newly created deployment object should contain both CloudProvider and PlatformProvider since its Application model contains both CloudProvider and PlatformProvider (after the migration run)
https://github.com/pipe-cd/pipecd/blob/master/pkg/app/piped/trigger/deployment.go#L96-L97

Only one case leaks from this strategy, that's users update the piped while that piped still handling deployments, but for simplicity, I think to make that deployment fail and create a new one instead or warn users to ensure all deployments are done before upgrade piped to this version is a better option (compare to keep support both CloudProvider and PlatformProvider). wdyt?

Reply:
> When reading the PR's title, I thought that this PR was only focused on the replacement in the Piped configuration.

It's not, in this PR, I want to make PlatformProvider specified in piped configuration works as CloudProvider configuration. It means that if some application is already registered to pipecd control-plane and the migration task is done, the piped with PlatformProvider (as well as CloudProvider since internally they are one while piped find the provider to deploy) name satisfied the application's platformProvider value should be deployed normally.

Reply:
> The newly created deployment object should contain both CloudProvider and PlatformProvider since its Application model contains both CloudProvider and PlatformProvider (after the migration run)
https://github.com/pipe-cd/pipecd/blob/master/pkg/app/piped/trigger/deployment.go#L96-L97

Nice point. Forgot that. 👍 

> Only one case leaks from this strategy, that's users update the piped while that piped still handling deployments, but for simplicity, I think to make that deployment fail and create a new one instead or warn users to ensure all deployments are done before upgrade piped to this version is a better option (compare to keep support both CloudProvider and PlatformProvider). wdyt?

That is fine. Totally agree.